storage/heed/src/receipt.rs

```diff
@@ -0,0 +1,71 @@
+use {
+    crate::generic::EncodableB256,
+    heed::types::{LazyDecode, SerdeBincode},
+    moved_blockchain::receipt::{
+        ExtendedReceipt, ReceiptQueries, ReceiptRepository, TransactionReceipt,
+    },
+    moved_shared::primitives::B256,
+};
+
+pub type EncodableReceipt = SerdeBincode<ExtendedReceipt>;
+
+pub const RECEIPT_DB: &str = "receipt";
+
+#[derive(Debug)]
+pub struct HeedReceiptRepository;
+
+impl ReceiptRepository for HeedReceiptRepository {
+    type Err = heed::Error;
+    type Storage = &'static heed::Env;
+
+    fn contains(&self, env: &Self::Storage, transaction_hash: B256) -> Result<bool, Self::Err> {
+        let transaction = env.read_txn()?;
+
+        let db: heed::Database<EncodableB256, LazyDecode<EncodableReceipt>> = env
+            .open_database(&transaction, Some(RECEIPT_DB))?
+            .expect("Database should exist")
+            .lazily_decode_data();
+
+        db.get(&transaction, &transaction_hash).map(|v| v.is_some())
+    }
+
+    fn extend(
+        &self,
+        env: &mut Self::Storage,
+        receipts: impl IntoIterator<Item = ExtendedReceipt>,
+    ) -> Result<(), Self::Err> {
+        let mut transaction = env.write_txn()?;
+
+        let db: heed::Database<EncodableB256, EncodableReceipt> = env
+            .open_database(&transaction, Some(RECEIPT_DB))?
+            .expect("Database should exist");
```

Comment:
nit: We can say `Receipt database should exist` to make it more specific. This applies to other `heed` PRs too.